### DISCUSSION TITLE
Support needed for MOVE

### DISCUSSION THREAD

**Discussion by rebeccaringuette** [spacepy/spacepy#726] [Design and roadmapping]
Community members that become a part of the Magnetopause Open Validation Project (MOVE, https://osf.io/v4drt/) will need to read, interpolate through, and visualize in 3-4 dimensions (at minimum 2D plots of one or more variables with any combination of coordinates, including time, and at best 3D animations) outputs from the SWMF and BATS-R-US models to compare those outputs with observational data of magnetopause crossings. This analysis is expected to take place on the cloud using data stored in s3 buckets on AWS-east. Recent experiments with SpacePy and other PyHC packages showed that such a capability is not yet possible for modeled outputs larger than a reasonable memory (a few GB) or for SWMF global magnetosphere outputs stored in s3. I ask this group to please consider working towards these capabilities to support not only researchers using the technology stack built for MOVE, but also for other researchers attempting similar tasks. It would also be helpful to create good documentation for researchers to demonstrate how to do these tasks both on a local machine and on the cloud (hopefully these will be nearly identical, but appreciably faster on the cloud). Maybe adding a cloud-optimized format to SWMF output or a conversion routine to make cloud-optimized SWMF output could be useful, and maybe dask is a possible complimentary way forward. The researchers will also need similarly high quality documentation guiding researchers through coordinate conversions using SpacePy and the link between SpacePy and AstroPy coordinates. Anything else this group can think of to help support magnetopause and magnetospheric research would be greatly appreciated (interactive notebooks in SpacePy documentation? Widgets or buttons to click instead of notebooks?).

Slava Merkin just presented at AGU getting ParaView running on the cloud for something similar for MAGE output, but I don't think ParaView is free (maybe?) but it is at least open source. Might be a good way forward if ParaView turns out to be free (and easy to install). The only other method I've seen that even got close on the visualization capabilities used PlotLy. the TIEGCMpy presentation at DASH 2023 mentioned that PlotLy wasn't enough for them, but Trame seemed to be working well at that point (same group as Slava Merkin). Slava is open to collaboration.

**Comment by jtniehof** [comment]
Thanks Rebecca. As a first hack at decomposing tasks here:

- I've opened up a discussion on [naive reading of cloud objects](https://github.com/spacepy/spacepy/discussions/727) to capture the various suggestions that have been made...we'll start working that.
- The actual magnetopause crossing work can be facilitated with the SWMF last closed field line support. I'm going to leave that to @spacecataz to open up (and link) an appropriate issue for what work needs to happen in pybats.
- Full cloud-ready analysis support for SWMF, including cloud-ready hdf5 output and pybats support, is something that should be driven from the SWMF side by @spacecataz . I had a conversation with someone from the HDF5 group on this topic and will happily contribute.
- Full 3D reading / interpolation / visualization should probably be taken up in conjunction with the cloud-ready formats.
- I think @spacecataz 's PyHC summer school examples show pybats works reasonably well in a notebook environment at this point, but if anybody runs into weirdness we should revisit...and documenting any new functionality by including notebook examples will help us catch that. I'll think about this some more--I'd like to make checking for notebook support a by-default thing for all new visualization functionality in SpacePy.

**Reply by rebeccaringuette** [reply]
Would access to a cloud platform (e.g. HelioCloud) with costs covered help support this work, or is that already available to the team?

**Reply by jtniehof** [reply]
HelioCloud access would definitely help!

**Reply by rebeccaringuette** [reply]
@spacecataz What do you think about collaborating with the MAGE group (e.g. Brent Smith) to get SWMF visualizations working on the cloud?

**Reply by rebeccaringuette** [reply]
Are there any updates on these ideas?
